.github/src/changed-files.js

```diff
@@ -0,0 +1,114 @@
+// @ts-check
+
+import { diff } from "./git.js";
+
+/**
+ * @param {import('github-script').AsyncFunctionArguments['core']} core
+ * @param {string} [baseCommitish] Defaults to "HEAD^".
+ * @param {string} [headCommitish] Defaults to "HEAD".
+ * @returns {Promise<string[]>} List of changed files, relative to the repo root.  Example: ["specification/foo/Microsoft.Foo/main.tsp"]
+ */
+export async function getChangedFiles(
```

Comment:
I would suggest adding another filter for removing deleted files from the set of changes as most tools working on files will not want to process the deleted files.

Reply:
Will address in future PR: https://github.com/Azure/azure-rest-api-specs/issues/33194